code/modules/mob/living/carbon/xenomorph/attack_alien.dm

```diff
@@ -75,13 +75,22 @@
 	if(!(signal_return & COMSIG_XENOMORPH_BONUS_APPLIED))
 		damage_mod += dam_bonus
 
+	var/armor_pen = X.xeno_caste.melee_ap
+	if(HAS_TRAIT(X, STEALTH_TRAIT)) //sneeky attack
+		var/datum/action/ability/xeno_action/stealth/can_sneak_attack = locate() in X.actions
+		if(can_sneak_attack)
+			var/datum/action/ability/xeno_action/stealth/sneak_attack_armor_pen = locate() in X.actions
+			armor_pen = sneak_attack_armor_pen
+			var/datum/action/ability/activable/xeno/hunter_mark/assassin/mark = X.actions_by_path[/datum/action/ability/activable/xeno/hunter_mark/assassin]
```

Comment:
you can just use actions by path instead of locate here everywhere

Reply:
hm well i had issues when i did that with mark stuff but alright i'll try and see if it works.

Reply:
i'll try, i feel like there was a reason for me to do that but i cant recall right now its been a while.